src/interface/make_pack_descriptor.hpp

```diff
@@ -0,0 +1,114 @@
+//========================================================================================
+// (C) (or copyright) 2020-2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef INTERFACE_MAKE_PACK_DESCRIPTOR_HPP_
+#define INTERFACE_MAKE_PACK_DESCRIPTOR_HPP_
+
+#include <algorithm>
+#include <functional>
+#include <limits>
+#include <map>
+#include <memory>
+#include <regex>
+#include <set>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "interface/mesh_data.hpp"
+#include "interface/meshblock_data.hpp"
+#include "interface/metadata.hpp"
+#include "interface/sparse_pack.hpp"
+#include "interface/state_descriptor.hpp"
+#include "mesh/mesh.hpp"
+
+namespace parthenon {
+
+inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<std::string> &vars,
```

Comment:
:+1: I approve of moving these. Worth noting though that this is a (very minor) breaking change for downstream codes. 

I was going to suggest putting this in the `parthenon/package.hpp` header file and sticking `MakePackDescriptor` in the parthenon package prelude namespace. But I see you beat me to it. :D

Reply:
Also, feel like sneaking
```C++
template <class... Ts>
inline auto MakePackDescriptor(SparsePack<Ts...> pack,
                               StateDescriptor *psd,                                
                               const std::vector<MetadataFlag> &flags = {},
                               const std::set<PDOpt> &options = {}) ) {
  return parthenon::MakePackDescriptor<Ts...>(psd, flags, options);
}
```
in here?

Reply:
added.

Reply:
I guess this is only breaking for Riot, right @Yurlungur?

Reply:
I think that's right. And it should be a pretty easy fix in riot, so I think the breaking change is worth it.

Reply:
Just was deciding if I should mark the PR as breaking.